retrofit/src/main/java/retrofit/MethodHandler.java

```diff
@@ -28,7 +29,9 @@
     Converter<ResponseBody, Object> responseConverter =
         (Converter<ResponseBody, Object>) createResponseConverter(method, retrofit, responseType);
     RequestFactory requestFactory = RequestFactoryParser.parse(method, responseType, retrofit);
-    return new MethodHandler<>(retrofit, requestFactory, callAdapter, responseConverter);
+    String infoForException = collectInfoForException(method);
```

Comment:
Actually doing that will allow you to pass it down into the `OkHttpCall` easily.